Alertmanager: Templatize config, user, and url.

pop opened the pull request:
@mburns @elijah (at any other maintainers)

This Pull Request adds the ability to template the http url, config destination, and user of the alert manager service(s).

Let me know if there's any edits I should make before it can be merged -- or if there are any fundamental problems with it being added in the first place. Like maybe we should add tests. Tests are cool. 😄

MarkGibbons commented on recipes/alertmanager.rb:
```diff
@@ -65,6 +65,9 @@
 
+node['prometheus']['flags']['alertmanager.url'] = 'http://127.0.0.1/alert-manager/' if \
+  node['prometheus']['flags']['alertmanager.url'].nil? || node['prometheus']['flags']['alertmanager.url'].empty?
+
```
Hi ElijahCaine,

  I'm a little worried about your code placement.  This looks like you want to set a default.  Why not just set the default in the attributes file?/

You will also need to specify a level for the value you set.  node.default[....  for instance.

Mark

pop replied:
Good catch. I totally brain-farted that code placement. Thanks!

MarkGibbons commented on templates/default/systemd/alertmanager.service.erb:
```diff
@@ -6,9 +6,11 @@ After=network.target
-    -web.external-url=http://127.0.0.1/alert-manager/
-User=prometheus
+    -config.file="<%= node['prometheus']['alertmanager']['config.file'] %>" \
+    -web.external-url="<%= node['prometheus']['flags']['alertmanager.url'] %>"
```
EC - you might want to add a test to verify that you did all of the attribute setting correctly and ended up with the value in the template that you expected.

pop replied:
I'll look into adding some tests for this. I didn't see *any* tests for Alertmanager in this repo so I might go ahead and just do some smoke tests for the whole thing while I'm at it.

pop committed:
Alertmanager: spec tests and better templating

Includes:
- Improved templating for systemd and bluepill configs.
- Spec and integration tests for alertmanager.
- Changes to attributes and service/pill files.

pop committed:
rubocop: conform to existing coding standards

pop commented:
I edited the `.rubocop.yml` config so it should conform with our coding standards, meaning we can run `rubocop -c .rubocop.yml` locally and it matches what houndci will complain about. I was just running `rake spec travis:ci` and it didn't hit any of those warnings.

Anyway, I also added a bunch of tests and fixed up the code a bit @MarkGibbons and @elijah (plus other contributors) so please let me know what else I can do to get this merged.

elijah commented:
Merged.  Thank you for all the work, and for the attention to tests and code style conventions!
